jax/numpy/lax_numpy.py

```diff
@@ -2277,11 +2277,16 @@ def atleast_3d(*arys):
 def array(object, dtype=None, copy=True, order="K", ndmin=0):
   if order is not None and order != "K":
     raise NotImplementedError("Only implemented for order='K'")
+  # check if the given dtype is compatible with JAX
   lax._check_user_dtype_supported(dtype, "array")
   dtype = dtype and dtypes.canonicalize_dtype(dtype)
 
   if _can_call_numpy_array(object):
     object = _np_array(object, dtype=dtype, ndmin=ndmin)
+    # check if the inferred type is compatible with JAX
+    _inferred_dtype = object.dtype and dtypes.canonicalize_dtype(object.dtype)
+    lax._check_user_dtype_supported(_inferred_dtype, "array")
+
```

Comment:
I would suggest putting this check under `if type(object) is np.ndarray`, as that is the place where the error can crop up.

I'm fairly certain that the choice to put it here is fine, because the only way you can end up in that lower block is if you are first in this block, but it would require less reasoning about the control flow if the check is put below.

Does that sound right to you?

Reply:
Sounds good! Made the requested change and rebased.